controllers/replication.storage/volumereplication_controller.go

```diff
@@ -203,17 +228,32 @@ func (r *VolumeReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 
 			return reconcile.Result{}, err
 		}
+		switch instance.Spec.DataSource.Kind {
+		case pvcDataSource:
+			err = r.annotatePVCWithOwner(ctx, logger, req.Name, pvc)
+			if err != nil {
+				logger.Error(err, "Failed to annotate PVC owner")
+				return ctrl.Result{}, err
+			}
 
-		err = r.annotatePVCWithOwner(ctx, logger, req.Name, pvc)
-		if err != nil {
-			logger.Error(err, "Failed to annotate PVC owner")
-			return ctrl.Result{}, err
-		}
+			if err = r.addFinalizerToPVC(logger, pvc); err != nil {
+				logger.Error(err, "Failed to add PersistentVolumeClaim finalizer")
 
-		if err = r.addFinalizerToPVC(logger, pvc); err != nil {
-			logger.Error(err, "Failed to add PersistentVolumeClaim finalizer")
+				return reconcile.Result{}, err
+			}
+		case volumeGroupReplicationDataSource:
+			err = r.annotateVolumeGroupReplicationWithOwner(ctx, logger, req.Name, vgr)
```

Comment:
Shouldn't VGR be the owner of VR?
I mean the VGR controller is creating the VR, and we will be setting controllerRef on the VR in the VGR controller, so any special benefit of using this?
Also, I am adding this annotation `replication.storage.openshift.io/volumegroupref` on both VR, and VGRContent created by VGR to fetch the VGR later on during reconcile.

Reply:
This is a safeguard to ensure that VR won't get deleted before deleting the VGR (it might be an extra check to make independent decision on VR controller not to depend on the VGR workflow)

Reply:
Ok, makes sense